FindANSAK.cmake

```diff
@@ -105,317 +108,357 @@ For DevStudio Generators, the picture is slightly murkier. See following comment
-    message("PROGRAM_DATA=\"$ENV{PROGRAM_DATA}/${PROGRAM_DATA}\"")
-    message("ANSAK_NO_GENERATOR_LIBS=$ENV{ANSAK_NO_GENERATOR_LIBS}/${ANSAK_NO_GENERATOR_LIBS}")
+if ( ANSAK_DEBUG )
+    message( "ANSAK_DIR=\"$ENV{ANSAK_DIR}/${ANSAK_DIR}\"" )
+    message( "PROGRAM_DATA=\"$ENV{PROGRAM_DATA}/${PROGRAM_DATA}\"" )
+    message( "ANSAK_NO_GENERATOR_LIBS=$ENV{ANSAK_NO_GENERATOR_LIBS}/${ANSAK_NO_GENERATOR_LIBS}" )
 endif()
 #
 ############################################################################
 
-set(win32Dirs Win32 x86 i386 X86)
-set(win64Dirs x64 Win64 x86-64 x86_64 X64 X86_64 X86-64 amd64)
-set(armDirs ARM Arm arm)
-set(arm64Dirs ARM64 Arm64 arm64)
-function(platform_matches_dirname dirname result)
-    if(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM")
-        set(theList ${armDirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM64")
-        set(theList ${arm64Dirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "Win32")
-        set(theList ${win32Dirs})
-    elseif(CMAKE_GENERATOR_PLATFORM STREQUAL "x64")
-        set(theList ${win64Dirs})
+set( win32Dirs Win32 x86 i386 X86 )
+set( win64Dirs x64 Win64 x86-64 x86_64 X64 X86_64 X86-64 amd64 )
+set( armDirs ARM Arm arm )
+set( arm64Dirs ARM64 Arm64 arm64 )
+function( platform_matches_dirname dirname result )
+    if( CMAKE_GENERATOR_PLATFORM STREQUAL "ARM" )
+        set( theList ${armDirs} )
+    elseif( CMAKE_GENERATOR_PLATFORM STREQUAL "ARM64" )
+        set( theList ${arm64Dirs} )
+    elseif( CMAKE_GENERATOR_PLATFORM STREQUAL "Win32" )
+        set( theList ${win32Dirs} )
+    elseif( CMAKE_GENERATOR_PLATFORM STREQUAL "x64" )
+        set( theList ${win64Dirs} )
     else()
-        message(WARNING "Did not find ${dirname} -- do you need to pass in its contents (dollar-curly-brace)?")
-        set(${result} False PARENT_SCOPE)
+        message( WARNING "Did not find ${dirname} -- do you need to pass in its contents (dollar-curly-brace)?" )
+        set( ${result} False PARENT_SCOPE )
     endif()
 
-    list(FIND theList ${dirname} index)
-    if(index GREATER_EQUAL 0)
-        set(${result} True PARENT_SCOPE)
+    list( FIND theList ${dirname} index )
+    if( index GREATER_EQUAL 0 )
+        set( ${result} True PARENT_SCOPE )
     else()
-        set(${result} False PARENT_SCOPE)
+        set( ${result} False PARENT_SCOPE )
     endif()
 endfunction()
 
-function(mirror_value var value doc)
-    string(TOUPPER "${var}" upper_var)
-    set(${upper_var} ${value} CACHE FILEPATH "${doc}")
+function( mirror_value var value doc )
+    string( TOUPPER "${var}" upper_var )
+    set( ${upper_var} ${value} CACHE FILEPATH "${doc}" )
 endfunction()
 
-function(find_win_lib find_var signal_library doc_string)
-    if(ARGN)
-        list(GET ARGN 0 cand)
-        if (ANSAK_DEBUG)
-            message("Provided extra value, ${cand} to be searched for ${signal_library}, to be noted in ${find_var}")
+function( find_win_lib find_var signal_library doc_string )
```

Comment:
Function for iterating through the different possible places the ansak components can be.